sdk/src/main/java/io/singularitynet/sdk/payment/Payment.java

```diff
@@ -19,4 +19,20 @@
      */
     void toMetadata(Metadata headers);
 
+    /**
+     * Special payment to designate that payment cannot be returned.
+     */
+    static final Payment INVALID_PAYMENT = new Payment() {
+
+        @Override
+        public void toMetadata(Metadata headers) {
+        }
+
+        @Override
+        public String toString() {
+            return "INVALID_PAYMENT";
+        }
```

Comment:
What if we have an InvalidPayment class implementing the Payment Interface instead of constant? With the current implementation, if we add any function to PayemntInterface we need to make sure that we add the same function in INVALID_PAYMENT.

Reply:
> if we add any function to PayemntInterface we need to make sure that we add the same function in INVALID_PAYMENT.

Not sure how it is different for the `InvalidPayment` class, which should also implement all the methods of the interface. `INVALID_PAYMENT` constant is equal to the anonymous `Payment` implementation which has the only instance. It is exact what I want for the invalid payment entity.

Reply:
My bad did not realize that you are creating same payment class.